pkg/restore/restore.go

```diff
@@ -510,6 +540,52 @@ func (ctx *context) restoreFromDir(dir string) (api.RestoreResult, api.RestoreRe
 	return warnings, errs
 }
 
+func (ctx *context) createNamespaceAfterTerminate(ns *v1.Namespace, logger *logrus.Entry) error {
+	timeout := time.NewTimer(ctx.resourceTerminatingTimeout)
```

Comment:
You can replace the timer & ticker with apimachinery's `wait.PollImmediate` or `wait.PollImmediateUntil` (if you need to plumb through a stop channel).